core/account-id/src/lib.rs

```diff
@@ -154,17 +93,14 @@ impl AccountId {
     /// use near_account_id::AccountId;
     ///
     /// let near_tla: AccountId = "near".parse().unwrap();
-    /// assert!(near_tla.is_top_level_account_id());
+    /// assert!(near_tla.is_top_level());
     ///
     /// // "alice.near" is a sub account of "near" account
     /// let alice: AccountId = "alice.near".parse().unwrap();
-    /// assert!(!alice.is_top_level_account_id());
+    /// assert!(!alice.is_top_level());
     /// ```
-    pub fn is_top_level_account_id(&self) -> bool {
-        self.len() >= MIN_ACCOUNT_ID_LEN
-            && self.len() <= MAX_ACCOUNT_ID_LEN
-            && self.as_ref() != "system"
-            && !self.as_ref().contains('.')
+    pub fn is_top_level(&self) -> bool {
+        !self.is_system() && !self.contains('.')
```

Comment:
This `self.contains` give me a pause with respect to `Deref` impl. 

Right now, the following code compiles: 

```rust
#[test]
fn hmm() {
    let near_tla: AccountId = "near".parse().unwrap();
    assert!(near_tla.is_top_level());

    let alice_app: AccountId = "app.alice.near".parse().unwrap();

    let hmm = alice_app.contains(&*near_tla);
}
```

Which I guess isn't *wrong*, but definitely looks a bit weird. Maybe stay conservative and don't do `Deref` immediatelly?

Reply:
I feel this is fine.. The coercion of `&AccountId` to `&str` is nice to have. We can pretty much treat an `AccountId` as a smart pointer to a memory location guaranteed to hold a valid NEAR Account ID so, the `&*near_tla` feels logical to me, although I think most people would probably just use `.as_str()` in that case.